server/src/main/java/org/elasticsearch/action/search/SearchPhaseController.java

```diff
@@ -258,42 +254,38 @@ TopDocs mergeTopDocs(Collection<TopDocs> results, int topN, int from) {
             final TopFieldDocs[] shardTopDocs = results.toArray(new TopFieldDocs[numShards]);
 
```

Comment:
Can you extract the logic below in a function that rewrites the field docs and the sort in place ?:
````
void transformNanoToMilli(TopFieldDocs[] shardTopDocs, Sort sort);
````
We need to apply the same logic to the `CollapseTopFieldDocs` above.

Reply:
done!